Input component

Noorieee opened the pull request:
Default:
![image](https://github.com/user-attachments/assets/b8de2bf6-53d6-4bfa-a2ef-a768339606bc)

Focused:
![image](https://github.com/user-attachments/assets/6fe13cdf-24b7-42cf-8748-9fac423037f1)

Error Message:
![image](https://github.com/user-attachments/assets/3f56be8f-06ed-4482-844a-f10a034824f3)

Disabled:
![image](https://github.com/user-attachments/assets/950b5228-8e6e-4d24-8ac8-0957cadfdab2)

Noorieee committed:
Created an input component

goralight commented on src/components/Atoms/Input/index.tsx:
```diff
@@ -0,0 +1,155 @@
+      position: absolute;
+      top: ${isFocused ? '0px' : '16px'};
+      left: ${isFocused ? '6px' : '16px'};
+      // transform: translate(-75%);
```
comment

goralight commented on src/components/Atoms/Input/index.tsx:
```diff
@@ -0,0 +1,155 @@
+  }
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    // Throwing onChange to parent to handle
```
technically not throwing
throwing is when you throw an error for the app to catch
this is just using the passed down function within your event handler

goralight commented on src/components/Atoms/Input/Input.stories.tsx:
```diff
@@ -0,0 +1,110 @@
+  args: {
+    label: 'Name',
+    isDisabled: false,
+    errorMessage: '',
```
i wouldnt specify `errorMessage` as an empty string, as this isnt really a legit use case example
if you arent setting it, i wouldnt add it to each story's args. same for isDisabled

Noorieee committed:
Removed commented css style
Updated handle OnChange comment
Removed disabled and error message args for stories that are not using it

Noorieee committed:
Updated folder structure and moved challenges inside a challenges folder to make space for future components

goralight commented on src/components/challenges/atoms/Input/index.tsx:
```diff
@@ -0,0 +1,154 @@
+    let backgroundColor = '#ffffff'
+    let outlineColor = '#6C6C6C'
+
+    if(isDisabled) {
```
should have a space between if and bracket

goralight commented on src/components/challenges/atoms/Input/index.tsx:
```diff
@@ -0,0 +1,154 @@
+
+    if(isDisabled) {
+      backgroundColor = '#E9E9E9'
+    } else if(hasErrorMessage) {
```
and here

goralight commented on src/components/challenges/atoms/Input/index.tsx:
```diff
@@ -0,0 +1,154 @@
+  }
+)
+
+const InputField = styled.input<{ isDisabled: boolean, hasErrorMessage: boolean, }>(
```
comma at the end isnt needed

goralight commented on src/components/challenges/atoms/Input/index.tsx:
```diff
@@ -0,0 +1,154 @@
+import React, { useState } from 'react'
+import styled from '@emotion/styled'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
```
dont need semi colons

goralight commented on src/components/challenges/atoms/Input/Input.stories.tsx:
```diff
@@ -0,0 +1,104 @@
+    label: 'Name',
+    isDisabled: true,
+    id: 'disabled',
+    value: 'Cannot edit this',
```
dont need comma at the end. check the rest of the file

goralight commented on src/components/challenges/atoms/Input/Input.stories.tsx:
```diff
@@ -0,0 +1,104 @@
+    const [value, setValue] = useState(initialValue)
+
+    const handleOnChange = (value: string) => {
+      setValue(value);
```
remove semi colon

goralight commented on src/components/challenges/atoms/Input/Input.stories.tsx:
```diff
@@ -0,0 +1,104 @@
+        defaultValue: { summary: 'text' },
+      },
+    },
+  },
```
dont need commas at the end of everything, should only have a comma if there is an item after it - if it is the last item, dont need comma

Noorieee committed:
Removed redundant semi colons
Added figma file link as parameter to story
Fixed spacing on conditional statement

goralight commented on src/components/challenges/atoms/Input/Input.stories.tsx:
```diff
@@ -0,0 +1,108 @@
+      }
+    }
+  },
+  render: ({ value: initialValue = '', ...args }) => {
```
you dont need this `value: initialValue = ''`, you can just make the initial value of the state below the empty string, like so:
```
const [value, setValue] = useState('')
```

Noorieee committed:
Removed value from render args and updated story state
